lib/openssl/ssl.rb

```diff
@@ -147,11 +147,22 @@ def set_params(params={})
         params = DEFAULT_PARAMS.merge(params)
         self.options = params.delete(:options) # set before min_version/max_version
         params.each{|name, value| self.__send__("#{name}=", value) }
+        
+        # Set the default certificate store if we're verifying certificates:
         if self.verify_mode != OpenSSL::SSL::VERIFY_NONE
           unless self.ca_file or self.ca_path or self.cert_store
             self.cert_store = DEFAULT_CERT_STORE
           end
         end
+
+        # Set the default session id context if it's not set:
+        unless self.session_id_context
+          # see #6137 - session id may not exceed 32 bytes
+          prng = ::Random.new($0.hash)
+          session_id = prng.bytes(16).unpack1('H*')
+          self.session_id_context = session_id
+        end
+
```

Comment:
`SSLContext#set_params` is useful only for TLS clients so the session ID context should not be set here (which only makes sense for servers).